pkg/operator/operator.go

```diff
@@ -216,6 +221,12 @@ func (optr *Operator) maoConfigFromInfrastructure() (*OperatorConfig, error) {
 		return nil, err
 	}
 
+	baremetalEnabled, baremetalImages, err := getAdditionalProviderImages(provider, *images)
```

Comment:
I was taking care of the scenario where the baremetal platform type is enabled but the images are not provided. I might have added some other checks along the same lines in other areas of the code. Thanks for pointing out that this cannot happen because the images will always be present.